src/mpi/ccl/cclcomm.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpiimpl.h"
+
+#define CUDA_ERR_CHECK(ret) if (unlikely((ret) != cudaSuccess)) goto fn_fail
+
+/*
+ * CCLcomm functions, currently tied to NCCL
+ */
+
+int MPIR_CCLcomm_init(MPIR_Comm * comm, int rank)
```

Comment:
To be consistent with the other CCLcomm functions, you might want to move the NCCL-specifics to a `MPIR_NCCL_` function.

Reply:
You can check how we managed PMI wrappers in `src/util/mpir_pmi.c` and `src/util/mpir_pmi[x].inc`

Reply:
All of the NCCL-specific code is now separated into its own file and activated by its own macro `ENABLE_NCCL`